bookstore/tests/test_handlers.py

```diff
@@ -1,6 +1,87 @@
 """Tests for handlers"""
+from unittest.mock import Mock
+
 import pytest
 
+from tornado.testing import AsyncTestCase
+from tornado.web import Application, HTTPError
+from tornado.httpserver import HTTPRequest
+from traitlets.config import Config
+
+from ..bookstore_config import BookstoreSettings
+from ..handlers import BookstoreVersionHandler, build_settings_dict
+
+
+@pytest.fixture(scope="class")
+def bookstore_settings(request):
+    mock_settings = {
+        "BookstoreSettings": {"s3_access_key_id": "mock_id", "s3_secret_access_key": "mock_access"}
+    }
+    config = Config(mock_settings)
+    bookstore_settings = BookstoreSettings(config=config)
+    if request.cls is not None:
+        request.cls.bookstore_settings = bookstore_settings
+    return bookstore_settings
+
+
+def test_build_settings_dict(bookstore_settings):
+    expected = {
+        'validation': {'archive_valid': True, 'bookstore_valid': False, 'publish_valid': True},
+        'version': '2.3.0.dev',
+    }
+    assert expected == build_settings_dict(bookstore_settings)
+
+
+@pytest.mark.usefixtures("bookstore_settings")
+class TestCloneAPIHandler(AsyncTestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.mock_application = Mock(
+            spec=Application,
+            ui_methods={},
+            ui_modules={},
+            settings={"bookstore": build_settings_dict(self.bookstore_settings)},
+            transforms=[],
+        )
+
+    def get_handler(self, uri, app=None):
+        if app is None:
+            app = self.mock_application
+        connection = Mock(context=Mock(protocol="https"))
+        payload_request = HTTPRequest(
+            method='GET',
+            uri=uri,
+            headers={"Host": "localhost:8888"},
+            body=None,
+            connection=connection,
+        )
+        return BookstoreVersionHandler(app, payload_request)
+
+    def test_get(self):
+        """This is a simple test of the get API at /api/bookstore
+
+        The most notable feature is the need to set _transforms on the handler.
+
+        The default value of handler()._transforms is `None`.
+        This is iterated over when handler().flush() is called, raising a TypeError.
+        
+        In normal usage, the application assigns this when it creates a handler delegate.
+
+        Because our mock application does not do this 
+        As a result this raises an error when self.finish() (and therefore self.flush()) is called.
+
+        At runtime on a live Jupyter server, application.transforms == [].
+        """
+        get_handler = self.get_handler('/api/bookstore/')
+        setattr(get_handler, '_transforms', [])
+        get_handler.get()
```

Comment:
We don't assert anything. There's a lot of code issues that would appear to pass this test when they are in fact not processing the get() call

Reply:
We're not doing this with all of the other handler tests. I agree it would be better, but I am not aiming at ideal right now.

Reply:
What are those code issues?

Reply:
What if the get returns None? Or get_handler returns the post handler? This test as is doesn't confirm much.

Reply:
I think the get _should_ return `None` given the way that tornado handlers are defined (it treats the finish as a side-effect that I'm not sure how to hook into). 

What it confirms is that the code runs without raising an error, meaning we've met the API for the `get` method, and that such a handler can be created without issue.

The problem is there is no way that I can seem to get access to whatever is sent over the wire based on get.